crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/sort.rs

```diff
@@ -32,10 +32,52 @@ impl ClassInfo {
         None
     }
 
-    /// Compare based on variants weight. Classes with higher weight go first.
+    /// Compare based on variants weight. Classes with lower weight go first.
+    /// First compare variants weight length. Only if their equal compare their actual weight.
     /// Returns `None` if they have the same weight.
-    fn cmp_variants_weight(&self, _other: &ClassInfo) -> Option<Ordering> {
-        // TODO: implement variant weight comparison.
+    fn cmp_variants_weight(&self, other: &ClassInfo) -> Option<Ordering> {
+        let current_weight = self.variant_weight.clone()?;
+        let other_weight = other.variant_weight.clone()?;
```

Comment:
Why do we need to `clone()`? This feels incorrect, because we are doing a comparison, and we don't need to own any information. If we do, I think it's a hint that something isn't correct.

Reply:
> Why do we need to `clone()`? This feels incorrect, because we are doing a comparison, and we don't need to own any information. If we do, I think it's a hint that something isn't correct.

Yeah it felt wrong, should `variant_weight` be a reference to that `BitVec` (and the same goes with `arbitrary_variants`?
I have a compiler error on them if I don't clone them